### DISCUSSION TITLE
How to update scripts for v2.0

### DISCUSSION THREAD

**Discussion by GinTonic99** [gerbera/gerbera#2950] [Q&A] [answered]
in the release notes for v2.0.0 I read:

_The JavaScript integration has be overhauled to simplify the process of providing additional layout scripts. If you created copies and modified that code to you purpose, you may have to update your code to the new interfaces for import functions._

Are there any instructions on how to update old scripts to match the new requirements?
thanks g.

**Comment by KarlStraussberger** [comment, marked as answer]
Situation with 1.12.1:
- global script `import.js` called some `add*` function (e.g. `addImage(obj)`)
- `import.js` was called as a whole and dispatched based on paramertes
- global variables `orig`, `object_script_path`, `object_autoscan_id` and `grb_container_type_...` were available in `add*` function

Situation with 2.0.0:
- Configuration of import-function per media type in `config.xml`. Import function is called directly during import process.
- No globals anymore
- Configuration options for `box-layout` to set the texts and options without tweaking the scripts.
- synopsis of the new `function importFunction(obj, cont, rootPath, autoscanId, containerType)`
- If you look at the new `import.js` file it tries to be compatible to some degree. Also the old config options are still available. Depending on your changes it may or may not work.

**Reply by GinTonic99** [reply]
thanks
